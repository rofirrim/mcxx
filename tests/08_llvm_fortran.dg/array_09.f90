! <testinfo>
! test_generator="config/mercurium-fortran-llvm run"
! test_stdout="array_09.f90.out"
! </testinfo>
MODULE MOO
    CONTAINS
SUBROUTINE SUB(A, B, C)
    IMPLICIT NONE
    REAL :: A(:), B(:), C(:)

    C = A + B
END SUBROUTINE SUB

END MODULE MOO

PROGRAM MAIN
    USE MOO
    IMPLICIT NONE
    INTEGER, PARAMETER :: N = 4
    REAL :: A(N), B(N), C(N)
    INTEGER :: I

    DO I = 1, N
      A(I) = I
      B(I) = I + 1
      C(I) = -I
    END DO

    CALL SUB(A, B, C)

    PRINT *, A
    PRINT *, B
    PRINT *, C
END PROGRAM MAIN
