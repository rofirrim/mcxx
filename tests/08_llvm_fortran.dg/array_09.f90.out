   1.00000000       2.00000000       3.00000000       4.00000000    
   2.00000000       3.00000000       4.00000000       5.00000000    
   3.00000000       5.00000000       7.00000000       9.00000000    
